chapters/ch06.md

```diff
@@ -1,45 +1,54 @@
 
 ->
+Vấn đề thứ hai với việc tập phát triển và tập kiểm tra có phân phối khác nhau: Có khả năng đội của bạn xây dựng được một mô hình hoạt động tốt trên tập phát triển nhưng lại kém trên tập kiểm tra. Tôi đã rất thất vọng và lãng phí nhiều thời gian với những mô hình như vậy. Hãy tránh để điều này xảy ra với bạn.  
 
 As an example, suppose your team develops a system that works well on the dev set but not the test set. If your dev and test sets had come from the same distribution, then you would have a very clear diagnosis of what went wrong: You have overfit the dev set. The obvious cure is to get more dev set data.
 
 ->
+Ví dụ, đội của bạn phát triển một hệ thống thực thi tốt trên tập phát triển nhưng kém trên tập kiểm tra. Nếu tập phát triển và tập kiểm tra có cùng một phân phối thì bạn có thể chuẩn đoán ngay vấn đề: Bạn đã overfit tập phát triển. Cách xử lý hiển nhiên nhất đó là bổ sung thêm dữ liệu cho tập phát triển.
 
 But if the dev and test sets come from different distributions, then your options are less clear. Several things could have gone wrong:
 
 ->
+Nhưng nếu tập phát triển và tập kiểm tra được tạo từ các phân phối khác nhau thì việc chuẩn đoán vấn đề sẽ phức tạp hơn. Các vấn đề có thể xảy ra là:   
 
 1. You had overfit to the dev set.
 
 ->
+Bạn overfit tập phát triển.
 
 2. The test set is harder than the dev set. So your algorithm might be doing as well as could be expected, and no further significant improvement is possible.
 
 ->
+Tập kiểm tra khó hơn tập phát triển. Thuật toán của bạn lúc này đã làm tốt nhất và không thể cải thiện thêm, hoặc cải thiện không đáng kể.  
 
 3. The test set is not necessarily harder, but just different, from the dev set. So what works well on the dev set just does not work well on the test set. In this case, a lot of your work to improve dev set performance might be wasted effort.
 
--> 
+->
+Tập kiểm tra có thể không khó hơn nhưng lại khác so với tập phát triển. Do đó mô hình có thể thực thi tốt trên tập phát triển nhưng lại kém trên tập kiểm tra. Trong trường hợp này việc cố gắng cải thiện tập phát triển có thể trở nên vô nghĩa.
 
 Working on machine learning applications is hard enough. Having mismatched dev and test sets introduces additional uncertainty about whether improving on the dev set distribution also improves test set performance. Having mismatched dev and test sets makes it harder to figure out what is and isn’t working, and thus makes it harder to prioritize what to work on.
 
 ->
+Nhiều vấn đề phức tạp có thể xảy ra khi làm việc với các ứng dụng machine learning. Tập phát triển và tập kiểm tra khác nhau khiến chúng ta không chắc chắn về việc nên cải thiện tập phát triển hay cải thiện tập kiểm tra cùng lúc. Tập phát triển và tập kiểm tra khác nhau cũng khiến cho việc chuẩn đoán vấn đề trở nên phức tạp hơn, nhất là trong việc lựa chọn các công việc ưu tiên cần thiết để cải thiện mô hình.
```

Comment:
Câu "Tập phát triển và tập kiểm tra khác nhau khiến chúng ta không chắc chắn về việc nên cải thiện tập phát triển hay cải thiện tập kiểm tra cùng lúc" khá lủng củng và không đúng nghĩa (mình không muốn nói là sai rất nhiều).

Đề xuất:
> Tập phát triển và kiểm tra không nhất quán dẫn đến sự không chắc chắn về việc liệu rằng cải thiện kết quả trên tập phát triển có giúp cải thiện kết quả trên tập kiểm tra. Sự không nhất quán này khiến việc tìm ra nguyên nhân và sắp xếp hướng giải quyết trở nên khó khăn hơn, vì không biết mô hình gặp vấn đề ở đâu.

Reply:
> Câu "Tập phát triển và tập kiểm tra khác nhau khiến chúng ta không chắc chắn về việc nên cải thiện tập phát triển hay cải thiện tập kiểm tra cùng lúc" khá lủng củng và không đúng nghĩa (mình không muốn nói là sai rất nhiều).
> 
> Đề xuất:
> 
> > Tập phát triển và kiểm tra không nhất quán dẫn đến sự không chắc chắn về việc liệu rằng cải thiện kết quả trên tập phát triển có giúp cải thiện kết quả trên tập kiểm tra. Sự không nhất quán này khiến việc tìm ra nguyên nhân và sắp xếp hướng giải quyết trở nên khó khăn hơn, vì không biết mô hình gặp vấn đề ở đâu.

- "performance" em thấy ở trên anh dịch là "hiệu quả", nên trong câu này "hiệu quả" sẽ chuẩn và thống nhất hơn.
>>Tập phát triển và kiểm tra không nhất quán dẫn đến sự không chắc chắn về việc liệu rằng cải thiện ~kết quả~ `hiệu quả` trên tập phát triển có giúp cải thiện ~kết quả~ `hiệu quả` trên tập kiểm tra. Sự không nhất quán này khiến việc tìm ra nguyên nhân và sắp xếp hướng giải quyết trở nên khó khăn hơn, vì không biết mô hình gặp vấn đề ở đâu.

Reply:
* "whether improving on the dev set distribution" ->  ở đây em nghĩ là nói về liệu việc thay đổi sự phân bố (phân phối) của dữ liệu trên tập dev set, chứ ko phải là performance

Reply:
@SumoBBQ nice catch. Em có thể tạo một PR mới sửa đoạn này được không. Chương này anh đã merge nhưng vẫn để toàn bộ tiếng Anh và chờ bạn dịch mượt hơn và chuẩn hơn.

Reply:
@tiepvupsu em giờ ko tiện lắm.
@quangnhat185 em làm ơn sửa hộ mọi người được không, giờ anh đang ko tiện lắm.